File: 1382A.cpp
#include <iostream>

using namespace std;

int main()
{
	int t, a_l, b_l;
	cin>>t;
	while(t--){
		cin>>a_l>> b_l;
		int a[a_l], b[b_l];
		for(int x=0; x<a_l; x++){
			cin >> a[x];
		}
		for(int x=0; x<b_l; x++){
			cin >> b[x];
		}
		bool found = false;
		for(auto x : a){
			for(auto y : b){
				if(x==y){
					found = true;
					cout<<"YES"<<endl;
					cout<< 1 << " " << x << endl;
					break;
				}
			}
			if(found){break;}
		}
		if(!found){
			cout<<"NO" <<endl;
		}
	}

    return 0;
}

File: 1382A.py
def common_elements(list1, list2):
    return [element for element in list1 if element in list2]

t = int(input())
while t>0:
    al, bl = [int(x) for x in input().split()]
    a = [str(int(x)) for x in input().split()]
    b = [str(int(x)) for x in input().split()]    
    found = False
    for x in a:
        if x in b:
            found = True
            print("YES")
            print(f'1 {x}')
            break
    if not found:print("NO")
    t-=1

